src/objects.c

```diff
@@ -2185,6 +2191,15 @@ static int match_public_keys(P11PROV_OBJ *key1, P11PROV_OBJ *key2)
         return cmp_public_key_values(key1, key2);
     }
 
+    if (key1->data.key.type == CKK_EC) {
+        /* Vendor implementation where EC private key may contain
+         * CKA_EC_POINT
+         */
+        ret = cmp_public_key_values(key1, key2);
+        if (ret != RET_OSSL_ERR)
+            return ret;
+    }
+
```

Comment:
Here we should probably just call cmp_public_key_values() unconditionally at the top of the function, if it succeed we return positively, if it doesn't we continue.
Returning an error is wrong because it will prevent tokens that work per spec *and* have an associated public key, from working.

Reply:
Clarification - this returns on success - (!= RET_OSSL_ERR) - if there is an error then if will fall-through to the find_associated_obj. I.e. leave early on success, on failure continue with the original business logic.

I will wait to rebase after the previous PR is completed.

Reply:
My bad, I read the check backwards, behavior is correct on error, the rest of the comment stands.
Ok to merge the other PR first.

Reply:
Squashed with the comments incorporated